toolchain/install/install_paths.h

```diff
@@ -103,35 +78,51 @@ class InstallPaths {
     return error_;
   }
 
-  // The computed installation prefix. This should correspond to the
-  // `prefix_root` directory in Bazel's output, or to some prefix the toolchain
-  // is installed into on a system such as `/usr/local` or `/home/$USER`.
-  //
-  // This will be an absolute path. We keep an absolute path for when the
-  // command line uses a relative path (`./bin/carbon`) and the working
-  // directory changes after initialization (for example, to Bazel's working
-  // directory).
-  //
-  // In the event of an error, this will be the empty string.
-  auto prefix() const -> llvm::StringRef { return prefix_; }
-
-  auto driver() const -> std::string;
+  // The directory containing the `Core` package. Computed on demand.
   auto core_package() const -> std::string;
+
+  // The directory containing LLVM install binaries. Computed on demand.
   auto llvm_install_bin() const -> std::string;
 
  private:
+  friend class InstallPathsTestPeer;
+
   InstallPaths() { SetError("No prefix provided!"); }
   explicit InstallPaths(llvm::StringRef prefix) : prefix_(prefix) {}
 
   // Set an error message on the install paths and reset the prefix to empty,
   // which should use the current working directory.
   auto SetError(llvm::Twine message) -> void;
 
-  // Check that the install paths have a marker file at the expected location,
-  // and if not calls `SetError` with the relevant error message.
+  // Check that the install paths have a marker file at
+  // `prefix()/lib/carbon/carbon_install.txt". If not, calls `SetError` with the
+  // relevant error message.
   auto CheckMarkerFile() -> void;
 
+  // The computed installation prefix. This will be an absolute path. We keep an
+  // absolute path for when the command line uses a relative path
+  // (`./bin/carbon`) and the working directory changes after initialization
+  // (for example, to Bazel's working directory). In the event of an error, this
+  // will be the empty string.
+  //
+  // When run from bazel (for example, in unit tests or development binaries)
+  // this will look like:
+  // `bazel-bin/some/bazel/target.runfiles/_main/toolchain/install/prefix_root`
```

Comment:
The `prefix_root` at the end looks a little suspicious to me. Is that right?

Reply:
We use an explicit `prefix_root` for all the things installed in the package to keep them distinct from the code and other things in the `install` package for managing an installation.